### ISSUE TITLE
🐞Expander

### ISSUE THREAD

**Issue by saladuit** [saladuit/minishell#37]

**Comment by saladuit** [comment]
https://stackoverflow.com/questions/5163144/what-are-the-special-dollar-sign-shell-variables/5163260#5163260
$1, $2, $3, ... are the [positional parameters](https://www.gnu.org/software/bash/manual/html_node/Positional-Parameters.html).
"$@" is an array-like construct of all positional parameters, {$1, $2, $3 ...}.
"$*" is the IFS expansion of all positional parameters, $1 $2 $3 ....
$# is the number of positional parameters.
$- current options set for the shell.
$$ pid of the current shell (not subshell).
$_ most recent parameter (or the abs path of the command to start the current shell immediately after startup).
$IFS is the (input) field separator.
$? is the most recent foreground pipeline exit status.
$! is the PID of the most recent background command.
$0 is the name of the shell or shell script.

**Comment by saladuit** [comment]
except $? everything should be invalid input

**Comment by saladuit** [comment]
Lexical analyzer should handle all invalid input including unclosed quotes, so that shouldn't be handled by expander

**Comment by saladuit** [comment]
```
 1:32:50 ~/repos/minishell
> echo $SHELL$SHELL
/bin/zsh/bin/zsh
 1:33:00 ~/repos/minishell
> echo $SHELL$SHELL$$$$
/bin/zsh/bin/zsh37793779
 1:33:12 ~/repos/minishell
> echo $SHELL$SHELL'$$'$$
/bin/zsh/bin/zsh$$3779
 1:33:24 ~/repos/minishell
> echo "$SHELL$SHELL'$$'$$"
/bin/zsh/bin/zsh'3779'3779
 1:33:36 ~/repos/minishell
```

**Comment by saladuit** [comment]
closed by #32
